plugins/modules/elasticsearch_info.py

```diff
@@ -0,0 +1,300 @@
+          type: str
+          sample: COMPLETED
+    snapshot_options:
+      type: complex
+      contains:
+        automated_snapshot_start_hour:
+          type: int
+          sample: 23
+    tags:
+      returned: always
+      type: dict
+      sample:
+        application: api
+        environment: prod
+    upgrade_processing:
+      type: bool
+      sample: false
+    vpc_options:
+      type: complex
+      contains:
+        availability_zones:
+          type: list
+          sample: [ eu-west-3a, eu-west-3b, eu-west-3c ]
+        security_group_ids:
+          type: list
+          sample: [ sg-0123456789abcdef0 ]
+        subnet_ids:
+          type: list
+          sample: [ subnet-0123456789abcdef0 ]
+        vpc_id:
+          type: str
+          sample: vpc-0123456789abcdef0
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+
+@AWSRetry.exponential_backoff()
+def list_elasticsearch_domain_names(client):
+    domain_names = client.list_domain_names()['DomainNames']
+    domain_names = list(map(lambda x: x['DomainName'], domain_names))
+    return domain_names
+
+@AWSRetry.exponential_backoff()
+def describe_elasticsearch_domains(client, domain_name=None):
+    domain_names = []
+
+    if domain_name:
+        domain_names.append(domain_name)
+    else:
+        domain_names = list_elasticsearch_domain_names(client)
+
+    return client.describe_elasticsearch_domains(DomainNames=domain_names)['DomainStatusList']
+
+@AWSRetry.exponential_backoff()
+def get_domain_tags(client, arn):
+    tags = client.list_tags(ARN=arn)['TagList']
+    tags = boto3_tag_list_to_ansible_dict(tags)
+    return tags
+
+def enrich_domain(client, domain):
+    domain = camel_dict_to_snake_dict(domain)
+    domain['tags'] = get_domain_tags(client, domain['arn'])
+
+    return domain
+
+def get_elasticsearch_domains(module, client, domain_name=None):
+    try:
+        domains = describe_elasticsearch_domains(client, domain_name)
+    except is_boto3_error_code('ResourceNotFound'):
+        return []
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't obtain Elasticsearch domain info")
+
+    return list(map(lambda domain: enrich_domain(client, domain), domains))
+
+def main():
+    argument_spec = dict(
+        domain_name=dict(required=False)
+    )
+
+    module = AnsibleAWSModule(argument_spec=argument_spec, supports_check_mode=True)
+    client = module.client('es')
```

Comment:
```suggestion
    client = module.client('es', retry_decorator=AWSRetry.jittered_backoff())
```
https://docs.ansible.com/ansible/latest/dev_guide/platforms/aws_guidelines.html#api-throttling-rate-limiting-and-pagination

We also have a decorator which makes it easier to switch out the retry configuration when appropriate.  (Often more helpful when modules manipulate resources).

You can then enable the retries on each client call with something like
```
     domain_names = client.list_domain_names(aws_retry=True)['DomainNames'] 
```

(aws_retry is optional and defaults to False)

Personally I have a slight preference for the jittered backoff: https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/ but the exponential is fine as is the @ mechanism for enabling them